pkg/guestagent/guestagent_linux.go

```diff
@@ -5,26 +5,88 @@ import (
 	"encoding/binary"
 	"errors"
 	"reflect"
+	"sync"
 	"time"
 
+	"github.com/elastic/go-libaudit/v2"
+	"github.com/elastic/go-libaudit/v2/auparse"
 	"github.com/lima-vm/lima/pkg/guestagent/api"
 	"github.com/lima-vm/lima/pkg/guestagent/iptables"
 	"github.com/lima-vm/lima/pkg/guestagent/procnettcp"
 	"github.com/sirupsen/logrus"
 	"github.com/yalue/native_endian"
 )
 
-func New(newTicker func() (<-chan time.Time, func())) Agent {
-	return &agent{
+func New(newTicker func() (<-chan time.Time, func()), iptablesIdle time.Duration) (Agent, error) {
+	a := &agent{
 		newTicker: newTicker,
 	}
+
+	auditClient, err := libaudit.NewMulticastAuditClient(nil)
+	if err != nil {
+		return nil, err
+	}
+	auditStatus, err := auditClient.GetStatus()
+	if err != nil {
+		return nil, err
+	}
+	if auditStatus.Enabled == 0 {
+		if err = auditClient.SetEnabled(true, libaudit.WaitForReply); err != nil {
+			return nil, err
+		}
+	}
+
+	go a.setWorthCheckingIPTablesRoutine(auditClient, iptablesIdle)
+	return a, nil
 }
 
 type agent struct {
 	// Ticker is like time.Ticker.
 	// We can't use inotify for /proc/net/tcp, so we need this ticker to
 	// reload /proc/net/tcp.
 	newTicker func() (<-chan time.Time, func())
+
+	worthCheckingIPTables   bool
+	worthCheckingIPTablesMu sync.RWMutex
+	latestIPTables          []iptables.Entry
+	latestIPTablesMu        sync.RWMutex
+}
+
+// setWorthCheckingIPTablesRoutine sets worthCheckingIPTables to be true
+// when received NETFILTER_CFG audit message.
+//
+// setWorthCheckingIPTablesRoutine sets worthCheckingIPTables to be false
+// when no NETFILTER_CFG audit message was received for the iptablesIdle time.
+func (a *agent) setWorthCheckingIPTablesRoutine(auditClient *libaudit.AuditClient, iptablesIdle time.Duration) {
+	var latestTrue time.Time
+	go func() {
+		for {
+			time.Sleep(iptablesIdle)
+			a.worthCheckingIPTablesMu.Lock()
```

Comment:
Is this whole mutex really needed? Afaict it is just "protecting" a bool and an int, both of which should be updated atomically anyways. And the whole thing is a rough heuristic anyways, so it doesn't even matter if it gets the values wrong in one iteration.

But I would at least like to understand if this is necessary for some reason.

Reply:
`go test -race` will raise errors if we remove the mutex (and we add race tests in future)

Reply:
Ok, that's good enough a reason :)